### ISSUE TITLE
Cannot map custom scalars to third party pointer types 

### ISSUE THREAD

**Issue by albrow** [99designs/gqlgen#1259]
### What happened?

Following the [guide here](https://gqlgen.com/reference/scalars/#custom-scalars-with-third-party-types) I am trying to implement a custom scalar type `BigNumber` which maps to `*big.Int` in the standard library. This results in compiler errors in the generated code.

```
validation failed: packages.Load: /Users/alex/programming/gomod/gqlgen-todos/graph/generated/generated.go:1909:9: cannot use scalars.UnmarshalBigNumber(v) (value of type *big.Int) as big.Int value in return statement
/Users/alex/programming/gomod/gqlgen-todos/graph/generated/generated.go:1913:34: cannot use v (variable of type big.Int) as *big.Int value in argument to scalars.MarshalBigNumber
```

Here is the problematic code:

```go
func (ec *executionContext) unmarshalOBigNumber2mathᚋbigᚐInt(ctx context.Context, v interface{}) (big.Int, error) {
	return scalars.UnmarshalBigNumber(v)
}

func (ec *executionContext) marshalOBigNumber2mathᚋbigᚐInt(ctx context.Context, sel ast.SelectionSet, v big.Int) graphql.Marshaler {
	return scalars.MarshalBigNumber(v)
}
```

<img width="1180" alt="Screen Shot 2020-07-22 at 1 53 07 PM" src="https://user-images.githubusercontent.com/800857/88227631-bb352c80-cc22-11ea-94fb-86afacf85481.png">

<img width="1027" alt="Screen Shot 2020-07-22 at 1 52 57 PM" src="https://user-images.githubusercontent.com/800857/88227594-b1abc480-cc22-11ea-838d-c5bad54a964a.png">

In my real application, we need to use `*big.Int` and cannot use `big.Int`. All functions and methods in the `math/big` package expect `*big.Int` so if my models use `big.Int` it makes it extremely hard to work with.

### What did you expect?

Since I have defined custom marshaler and unmarshaler functions, I should be able to map `BigNumber` to `*big.Int` types in my models.

The models are actually being generated correctly. Here's what __model/models_gen.go__ looks like:

```go
type Todo struct {
	Number *big.Int `json:"number"`
}
```

The problem just lies in the generated code in __graph/generated/generated.go__. It looks like this can be resolved by changing the generated code so that it uses the correct types when calling `scalars.UnmarshalBigNumber` and `scalars.MarshalBigNumber`. I would suggest something like this:

```go
func (ec *executionContext) unmarshalOBigNumber2mathᚋbigᚐInt(ctx context.Context, v interface{}) (big.Int, error) {
	res, err := scalars.UnmarshalBigNumber(v)
	if err != nil || res == nil {
		return big.Int{}, err
	}
	return *res, err
}

func (ec *executionContext) marshalOBigNumber2mathᚋbigᚐInt(ctx context.Context, sel ast.SelectionSet, v big.Int) graphql.Marshaler {
	return scalars.MarshalBigNumber(&v)
}

func (ec *executionContext) unmarshalOBigNumber2ᚖmathᚋbigᚐInt(ctx context.Context, v interface{}) (*big.Int, error) {
	return scalars.UnmarshalBigNumber(v)
}

func (ec *executionContext) marshalOBigNumber2ᚖmathᚋbigᚐInt(ctx context.Context, sel ast.SelectionSet, v *big.Int) graphql.Marshaler {
	return scalars.MarshalBigNumber(v)
}
```

### Minimal graphql.schema and models to reproduce

See https://github.com/albrow/gqlgen-todos.

Schema:

```graphql
scalar BigNumber

type Todo {
  number: BigNumber
}
```

Marshal and unmarshal functions:

```go
package scalars

import (
	"fmt"
	"io"
	"math/big"
	"strconv"

	"github.com/99designs/gqlgen/graphql"
)

// UnmarshalBigNumber converts scalar value of type BigNumber into a *big.Int.
func UnmarshalBigNumber(v interface{}) (*big.Int, error) {
	s, ok := v.(string)
	if !ok {
		return nil, fmt.Errorf("BigNumber must be a numerical string")
	}
	bigInt := new(big.Int)
	_, ok = bigInt.SetString(s, 10)
	if !ok {
		return nil, fmt.Errorf("invalid BigNumber value: %q", s)
	}

	return bigInt, nil
}

// MarshalBigNumber converts a *big.Int into a scalar value of type BigNumber.
func MarshalBigNumber(number *big.Int) graphql.Marshaler {
	return graphql.WriterFunc(func(w io.Writer) {
		quotedString := strconv.Quote(number.String())
		_, _ = w.Write([]byte(quotedString))
	})
}
```

### versions
 - `gqlgen version`? `v0.11.3`
 - `go version`? `go version go1.14.3 darwin/amd64`
 - dep or go modules? go modules

**Comment by austincollinpena** [comment]
Did you find a solution to working with marshalling pointers?

**Comment by albrow** [comment]
Depends on what you mean by "find a solution". IMO gqlgen should not produce code that doesn't compile, and what I shared should be considered a bug.

For my specific application I was able to work around this bug by defining my own `BigNumber` type and writing the `MarshalGQL` and `UnmarshalGQL` methods. This creates unnecessary complication because we have to convert between the `BigNumber` type and `*big.Int` in order to do any actual math or business logic.
